src/Raven.Server/Documents/Subscriptions/SubscriptionConnectionState.cs

```diff
@@ -66,8 +78,16 @@ public override string ToString()
                                     $"Subscription {incomingConnection.Options.SubscriptionName} is already occupied by a TakeOver connection, connection cannot be opened");
 
                             if (_currentConnection != null)
-                                _storage.DropSubscriptionConnection(_currentConnection.SubscriptionId,
-                                    new SubscriptionInUseException("Closed by TakeOver"));
+                            {
+                                if (_isSharded)
+                                {
+                                   await _shardedContext.DropSubscriptionConnectionAndPropagateToShards(_currentConnection.SubscriptionId, new SubscriptionInUseException("Closed by TakeOver"));
```

Comment:
why do we need to propogate that? We close our connection, the connection to the shards should just be closed at that point, no?